### ISSUE TITLE
TypeError related to observer.horizon on next_pass()

### ISSUE THREAD

**Issue by adamkalis** [brandon-rhodes/pyephem#176]
Here is a code example that throws the TypeError that follows.

It seems that this error is related with the observer.horizon. For example, if you decrease the observer.horizon on the example bellow to 30 then it returns the next pass.

```
import ephem
import math

observer = ephem.Observer()
observer.lon = str(23.739)
observer.lat = str(38.048)
observer.elevation = 119
observer.horizon = "40"
observer.date = str('2020-07-29 08:46:32')

print(observer)

satellite = ephem.readtle(
    str('0 FOSSASAT-1'),
    str('1 44829U 19084F   20210.09398576  .00055392  00000-0  28444-3 0  9990'),
    str('2 44829  96.9680  79.4786 0025577  82.9270 277.4906 15.80589935 36852'))

satellite.compute(observer)

tr, azr, tt, altt, ts, azs = observer.next_pass(satellite, True)
print('\nObservation:')
print('Start: ', tr, 'End: ', ts, 'Elevation: ', format(math.degrees(altt), '.0f'))
```

The TypeError is:

```
Traceback (most recent call last):
  File "main.py", line 20, in <module>
    tr, azr, tt, altt, ts, azs = observer.next_pass(satellite, True)
  File "/opt/virtualenvs/python3/lib/python3.8/site-packages/ephem/__init__.py", line 532, in next_pass
    if result[0] <= result[2] <= result[4]:
TypeError: '<=' not supported between instances of 'NoneType' and 'NoneType'
```

**Comment by brandon-rhodes** [comment]
Well, drat — I am not able to reproduce that locally to help fix it. My computer says:

```
<ephem.Observer date='2020/7/29 08:46:32' epoch='2000/1/1 12:00:00' lon='23:44:20.4' lat='38:02:52.8' elevation=119.0m horizon=40:00:00.0 temp=15.0C pressure=1010.0mBar>
```

— when I run your script. What version of PyEphem, and on what kind of system are you?

**Comment by adamkalis** [comment]
I've created a repl.it with this example: https://repl.it/repls/EnchantedAdoredBracket#main.py

Linux with python version 3.8.3 and ephem version 3.7.7.1.

**Comment by brandon-rhodes** [comment]
Thanks, I've never used repl.it before! This is the first time that I haven't had to set my own machine up to run an example from a GitHub issue, it's a nice experience.

I pasted an alternate version of `next_pass()` into repl.it and then tried generating a plot on my system. Here is how the satellite behaves on the evening you are looking at:

![image](https://user-images.githubusercontent.com/166162/89477029-2e2dbf80-d75a-11ea-9347-279bb5277d0c.png)

The underlying C code that this calls is in `riset_cir.c` at:

https://github.com/brandon-rhodes/pyephem/blob/6849cc42dbb52284f9365655ba84cac5497de1f1/libastro-3.7.7/riset_cir.c#L160

It looks like the C code might be attempting Newton's method, to find the maximum, but maybe arrives at the maximum between 8pm and 9pm on the plot, which isn't high enough to actually bring it above the horizon? It's been a while since I looked at this old libastro code, so I don't remember if it has a way to escape a local maximum once it gets caught in one.

If you have a chance to try it out, I myself wrote the satellite pass finder in the new Skyfield library, and it tends to not have the limitations of the old `libastro` C routines which I wrapped up in Python to make PyEphem.

Anyway, feel free to look at `e_riset_cir()` more closely and see if there's some way we can protect against its misbehavior by wrapping it more intelligently in the Python code!
